### DISCUSSION TITLE
Release 1.2.0 Beta 13

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#1017] [General]
## PLUGIN

- Improved server management (resolves #1014)
- Added realtime/live mode (strict internal timeouts)
- Added more plugin monitor metrics
- Plugin search window improvements
- Fixed a backwards compatibility issue (fixes #1008)
- Stability fixes

## SERVER

- **NEW** Added option to autostart multiple servers (resolves #998)
- Improved scanning of multi-plugin shells
- Minor improvements and stability fixes

All downloads are available at https://audiogridder.com/download/.

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta13'>Release 1.2.0 Beta 13</a>.</em>

**Comment by emack999** [comment]
"Improved scanning of multi-plugin shells"

Still some issues loading the larger Waves plugin shells.  I can see that the scanning of the shells is happening (the individual plugin names inside the shell are being shown as they scanned), but it still prompts me to Wait or Cancel many times before the shell scanning is completed.

EDIT: It appears that the scan of the vst3 version of the Waves shells fails entirely.

**Comment by CaBuffalo** [comment]
Yes, this happens on Waves plugins. It’s a normal behaviour during the scanning.

**Comment by musicgreator** [comment]
With Beta12 and 13, i pretty much can't load any plugin, especially AcusticaAudio, but others also don't work. One showed me the GUI very briefly and then also gave me the error message. The message is: Failed to add XXX plugin   Error: seems like the plugin crashed the server or did not load (E_TIMEOUT). I deinstalled the plugin before installing Beta13, and whiped the cache/rescanned the plugins. Last working Version was Beta6 (only issue is, that there seems to be a memory limit at 10GB or so, i reported it before)

**Reply by musicgreator** [reply]
AudioGridder VST3 is working, in the VST3 version i can also select Stereo instead of Mulit-Mono only in the VST2 version. But also Multi-Mono works in the VST3 version. I'm using Reaper on Windows 10.

**Comment by djniviro** [comment]
For me an amazing update! Only one more feature towards the Beta14 would be channel linking when used in multi-mono mode. Ultimately it would be as follows:

- Next to the A/B and "power" button there should be a small "Link All" button (might be represented with a "chain" symbol), when pressed all multi-mono instances get dialed in the same settings when editing a setting. 
- When this "Link All" button is activated, all channels in the channel selector (currently next to the power button) should have a "tick" next to the corresponding channel which is activated. Natively all channels have to be linked when the Link All button has been pressed.
- A multi-mono instance should also be able to be unlinked (let's say LFE in multichannel mixes). An additional button "Unlink Instance" (might be represented with a "broken chain" symbol) could make this happen. When this Unlink Instance button has been pressed, the "tick" in the channel list disappears.

I know it might be a lot to ask, but if this were there, you'd be a saviour!

**Reply by apohl79** [reply]
Added a feature request for this. Thx for the suggestion!

**Reply by djniviro** [reply]
Awesome! Thank you so much!

**Comment by djniviro** [comment]
Oh also I seem to have a slightly offset with the mouse (but that has always been within any AudioGridder version). Is there like a way to enable an offset corrector? Hard values representing coordinates would do the job for sure!

**Reply by apohl79** [reply]
Interesting. Are you running a mac or windows server?

**Reply by djniviro** [reply]
I’m running a Windows server. But I experienced kinda like the same thing when running it on the Mac Mini M1 too. The offset is there on both the Windows and Mac plugin vst. Btw, it might sound like a stupid question, but is it correct that, in order to make the plugin work, I need to open a separate remote desktop window of the rendering machine on the background? Not that this is an issue for me, just trying to figure out the offset thing.

**Reply by apohl79** [reply]
In case you are using different DAWs, is the behaviour the same everywhere? I have not seen those offsets, so I'm wondering if they are caused on the client side, especially as you see them both on windows and mac servers. The local coordinates on the daw plugin screen are translated on the server side.

Manual offset correction values could be added, but I'm wondering what is causing the issue on your system.

**Reply by djniviro** [reply]
The behaviour is identical using Ableton and Nuendo on my Mac and Nuendo on my Windows PC. I've also used different VNC clients like the built in with Windows and the Remote Desktop app from Microsoft. Both result in the same amount of offset. I do have to say I experience a bit more offset with the Mac server. It's only vertical offset. I have to aim slightly lower with the mouse to adjust the parameter.

Also, I experience that the GUI is shown at a 150% size compared to the original (running on the Remote Desktop). However when adjusting the size of the plugin to let's say 75%, the size of the remote plugin gets normal, but the plugin chain/information gets too small to work easily. Perhaps also an extra feature to implement (gui size and plugin size).

**Reply by apohl79** [reply]
> Btw, it might sound like a stupid question, but is it correct that, in order to make the plugin work, I need to open a separate remote desktop window of the rendering machine on the background?

To capture the screen there must be a UI session running on macOS or Windows. This can either be done by simply logging in on a system with a monitor connected or by using VNC for instance. So you don't have to open a remote desktop connection, if you are just logged in on your server computer.

**Reply by apohl79** [reply]
> Also, I experience that the GUI is shown at a 150% size compared to the original (running on the Remote Desktop). However when adjusting the size of the plugin to let's say 75%

Sounds like you have scaling enabled on Windows?

**Reply by djniviro** [reply]
Beginner question: if there’s no remote connection made (thus only logged in with a local screen connected) between the 2 workstations, how will the capture work within the vst as there is no place to fill in login credentials? Or is the login already made using the agridder server app?

also, scaling is not possible when using a remote desktop, but I can confirm it looks equally to a 150% scaled version on my 4k monitor. No possibility to turn it back to 100% when using an external desktop connection

**Comment by musicgreator** [comment]
So with Beta 12 and Beta 13, VST2 can't load my plugins (E_TIMEOUT). Only VST3 works.

**Reply by djniviro** [reply]
I have the same thing when selecting a mono or multi mono instance of a vst plugin which only is written to be stereo

**Reply by musicgreator** [reply]
Ahh, that seems to be the problem.

**Comment by fojerhar** [comment]
Beta 13 working so far so good here (now having an old session opening issue, but It will be solved I m sure)
Where can I find that what should I learn from this beautiful new numbers and colors here?
![Screen Shot 2022-09-13 at 12 40 22](https://user-images.githubusercontent.com/72139943/189886423-3572cf26-5051-4663-965d-7175c4722413.png)

**Reply by musicgreator** [reply]
Doesn't look so good, many of the chains seem not to work (not connected, not loaded). At the top right, you see a decription for the colors. Are you using AudioGridder in the VST2 or VST3 format?

**Reply by fojerhar** [reply]
Solved the opening issue, the caustic Spacecontrol plugin made the collapse (still) it makes something weird, but I will use this one in aax mode simply . It collapses the session while opening it, but I can enable this single one if the session is opened.... so I have to save the session as it bypassed..... 
but anyhow this is not important now

**Reply by fojerhar** [reply]
vst3, but everything is working and doing their job with the audio

**Reply by musicgreator** [reply]
Oh i confused the colors of the dot and the ms values. The green dot indicates, everything is ok.

**Reply by musicgreator** [reply]
Could you please try AudioGridder VST2? VST2 doesn't work here, would be interresting, if that issue is the same on other systems.

**Reply by apohl79** [reply]
> Where can I find that what should I learn from this beautiful new numbers and colors here?

yeah.. had no time to update the documentation yet. in short:

Bav = average read queue level
B95 = 95th percentile of the read queue level
RdErr = read errors (mostly relevant if you turn on realtime mode, as this shows when AG is dropping audio)
Perf Proc = the time spent in the DAW audio thread for processing a buffer of audio passed from the DAW to AG (avg over the last 32 calls)
Perf Net = the time spent sending and receiving one block of audio to/from the server (avg over the last 32 blocks)

Colors: 
white=the time is <50% of the realtime window (eg: with 48kHz and 1024 samples block size, the rt window is ~21ms)
yellow=the time is <70%
orange=the time is <80%
red=the time is >=80%

**Reply by fojerhar** [reply]
I dropped in some vst2 into an existing session, and they was doing their job as they should, so vst2 is working here

**Reply by fojerhar** [reply]
This helps a lot where should I solve the issues, really useful (and looks sexy the bunch of numbers) 
thank you
So the red means I m over the buffering time window and have to be careful with that chain. You already answered yes, that 10ms is the window size at 96K, but it works!

**Reply by apohl79** [reply]
> So the red means I m over the buffering time window and have to be careful with that chain. You already answered yes, that 10ms is the window size at 96K, but it works!

Yeah, those chains in red are at risk to drain the read queue empty (you will see it in Bav/B95). Once this happens, the Perf Proc value will go up, blocking PT -> causing the performance error OR AG start dropping audio, when you enable the RT mode, which will make the RdErr value go up.

**Reply by fojerhar** [reply]
Did I miss something or do I have to enable RealTime mode somewhere?

**Reply by apohl79** [reply]
It is a new setting in the plugin under "Transfer Audio/MIDI".

**Reply by fojerhar** [reply]
with a session I always have a problem with one plugin cannot load(caustic opaleq) but the "reload all unloaded chain" button solves this beautifuly :)
thanks

**Reply by fojerhar** [reply]
In protools offline bounce became much much faster (from x0.7 now x2.0) is it possible?

**Reply by apohl79** [reply]
With RT mode on? I think it should not be enabled for offline bouncing.

**Reply by fojerhar** [reply]
> With RT mode on? I think it should not be enabled for offline bouncing.

No (as i remember well) with standard mode it became much faster. 
I was playing with RT for a while but did not notice any difference yet. 

At first during bounce some chains were collapsed, but with the new trey app infos I saw where the problem is, so splitted up the chain into two parts, and voila, problem solved, no more collapses (it was a long and high ms delay chain). 
Things can be done more conscious
Thanks

**Reply by apohl79** [reply]
got it, cool! and yeah, RT does not make anything faster. It just drops audio to respond in time.

**Comment by musicgreator** [comment]
There is only the Multi-Mono available in AudioGridder Beta13 VST2.
I tried it with different hosts, in VST2 there is only Multi-Mono, and the plugins can't be loaded. 
![Capture](https://user-images.githubusercontent.com/30467042/190478615-506a0f4d-a466-4732-afc7-2e3f9e5a742a.PNG)
![Capture 2](https://user-images.githubusercontent.com/30467042/190478824-b5b9dcfc-264e-4c10-83b7-e7d4542456b0.PNG)

**Reply by fojerhar** [reply]
Hi
Here seems working fine magenta both vst2 and vst3 also with beta13
![Screen Shot 2022-09-17 at 16 18 08](https://user-images.githubusercontent.com/72139943/190861755-f5b10991-0b83-4728-9d5c-c53c4e0354be.png)
Seems you have a weird extreme big latency, I don't know if it is normal

**Reply by musicgreator** [reply]
Not sure we are talking about the same thing. I'm not talking about VST2 and VST3 plugins within AudioGridder, but i mean the AudioGridder instance itself in VST2 format. 

(When i use a VST3 AudioGridder instance, everything is fine here too, also with VST2 plugins inside of AudioGridder VST3)

**Reply by musicgreator** [reply]
I set the large latency/buffer on purpose. It is needed for large projects with many plugins in series, to prevent audio dropouts. In small projects it is counterproductive of courese.

**Reply by fojerhar** [reply]
Ah, okay, in prottols I have only the aax format

**Reply by apohl79** [reply]
Since multi-mono support was added, plugin layouts are matched agains the channel config of the AG plugin. The issue with the VST2 version of AG in reaper seems to be, that reaper is always allocating all 32 in/out channels and AG will show only plugins which support mono (for multi-mono) and 32 outputs. This does not make much sense, so I will allow for the plugins default I/O config as well (in the next beta), which was the behaviour of previous AG versions.

**Reply by djniviro** [reply]
Perhaps a good addition would be: if a default plugin (or stereo) gets detected, the Audiogridder automatically adjusts its routed channels from x to stereo or the default config of the plugin. Then there will never be a mismatch in communication.

**Reply by apohl79** [reply]
Yeah, that seems like a good idea for empty chains, but in case you have already loaded plugins, the channel config can't be changed. For now I would keep this untouched and just add a "Default" button for any plugin that does not support the same number of channels as the AG plugin itself is currently running with.

**Reply by musicgreator** [reply]
Sounds great.
With AudioGridder Beta13 VST3, i didn't get the E_TMEOUT error. The plugins load normal and quick. Only with VST2 plugins can't be loaded.

**Reply by musicgreator** [reply]
Presumably the E_TIMEOUT happens, because the plugins aren't compatible with the Mulit-Mono configuration, so when Stereo is available in the next beta in VST2, this could  also solve this problem.

**Comment by bharder** [comment]
Hi there,
I have two Mac mini M1 as servers. My main machine is a IMac Pro Intel. 
When I recall a mix AG does not distribute the plugins as I did, but loads ALL instances to one server.
But the AG plugin shows the Server I had chosen in the first place. 
So if I distributed 15 instances on Mac Mini 1 and 15 to MMini 2, when I reopen all 30 are on Mac Mini 1. 
Any help is much appreciated!

**Reply by apohl79** [reply]
I'm not able to reproduce this. The unique id of the server a plugin is connected to is stored in the session/project of your DAW and restored when loading. Not sure whats going wrong. Could you provide your plugin log files? Just store the session in your DAW with the intended state,  re-open the session (which loads with the wrong state) and zip the log dirs (see https://audiogridder.com/bug-report-diagnostics-locations/).

**Comment by bharder** [comment]
One more thing….The plugin tray crashes after a few seconds, when I open the program and the Mmonitor does not show up, when I try to open it in the plugin. Is there an uninstaller? I tried reinstalling a few times, but no luck:(

**Reply by apohl79** [reply]
You can manually delete AG using your terminal app:

```
sudo rm -rf /Library/Audio/Plug-Ins/*/AudioGridder*
sudo rm -rf /Library/Application\ Support/Avid/Audio/Plug-Ins/AudioGridder*
sudo rm -rf /Applications/AudioGridder*
```

**Comment by bharder** [comment]
Hi,
Wow! Thank you soooo much for reaching back to me! I really apreciate it!
So here is the log file

One question
I do not understand the server ID. Is it correct, that I only need it to run a server locally or different ones locally, but don’t need it on the separate machines?
I tried adding server IDs but it didn’t help.
My setup is:
Main: ImacPro Intel 10GB
 to:  Netgear Switch to 2 Mac mini M1s.
BTW. You are German right? Dann könnte ich mir das englisch eigentlich sparen:)
Vielen Dank für alle Mühe!
Bernhard

**Reply by apohl79** [reply]
You have to use your browser to upload files to github (email attachments do not work).

And yes, I'm German, but let's keep the discussion in english here, so people can follow. ;-)

**Comment by bharder** [comment]
[Archiv 2.zip](https://github.com/apohl79/audiogridder/files/9607430/Archiv.2.zip)
Sorry....here are the files

**Reply by apohl79** [reply]
The best is when you just zip the whole directory. You just added the two "lastest" links, but the actual files are missing.

**Comment by bharder** [comment]
[AudioGridder.zip](https://github.com/apohl79/audiogridder/files/9615326/AudioGridder.zip)
alright, here wo go...hopefully

**Reply by apohl79** [reply]
Found the bug. When starting a server with a new ID the first time, it inherits the full default config - unfortunately including the unique server ID.

You can fix this manually for now:
1. stop the servers
2. open the config files of your servers (eg audiogridderserver3.cfg for server with ID 3, locations: https://audiogridder.com/server/#config)
3. remove the line `"UUID": "<your UUID>",` from each config and save them
4. start the server (each server will generate a new ID now)
5. open your session and reconnect, the assignments should be persistent now

**Reply by bharder** [reply]
Thank you so much for looking into this!! I am not at the studio today, but I will report back asap!
